galley/pkg/config/analysis/analyzers/virtualservice/gateways.go

```diff
@@ -70,5 +73,45 @@ func (s *GatewayAnalyzer) analyzeVirtualService(r *resource.Instance, c analysis
 
 			c.Report(collections.IstioNetworkingV1Alpha3Virtualservices.Name(), m)
 		}
+
+		if !vsHostInGateway(c, gwFullName, vs.Hosts) {
+			m := msg.NewVirtualServiceHostNotFoundInGateway(r, vs.Hosts, vsName.String(), gwFullName.String())
+
+			if line, ok := util.ErrorLine(r, fmt.Sprintf(util.VSGateway, i)); ok {
+				m.Line = line
+			}
+
+			c.Report(collections.IstioNetworkingV1Alpha3Virtualservices.Name(), m)
+		}
 	}
 }
+
+func vsHostInGateway(c analysis.Context, gateway resource.FullName, vsHost []string) bool {
+	var gatewayHost []string
+
+	c.ForEach(collections.IstioNetworkingV1Alpha3Gateways.Name(), func(r *resource.Instance) bool {
+		if r.Metadata.FullName == gateway {
+			s := r.Message.(*v1alpha3.Gateway)
+
+			for _, v := range s.Servers {
+				gatewayHost = append(gatewayHost, v.Hosts...)
+			}
+		}
+
+		return true
+	})
+
+	for _, gh := range gatewayHost {
+		if gh == util.Wildcard {
+			return true
+		}
+
+		for _, vsh := range vsHost {
+			if vsh == util.Wildcard || vsh == gh {
```

Comment:
We support partial wildcards like `*.example.com` which would not match `foo.com`

Reply:
actually I forget this one

Reply:
it will apply to both, right?

for example:
- Gateway with `*.example.com` and VirtualService with `testing.example.com` true
- Gateway with `testing.example.com` and VirtualService with `*.example.com` true

Reply:
also, there are use cases like:
- Gateway `*.example.com` and the VirtualService is `web.testing.example.com`. Is this the correct configuration or not?

Reply:
great question. 

I believe `Gateway *.example.com and the VirtualService is web.testing.example.com` will match per implementation in https://github.com/istio/istio/blob/65bd2ac632cbd6ec8e3fd85bb9874f685aea0cfd/pkg/config/host/names.go#L78. I would agree it *shouldn't* match, but it does.

>Gateway with *.example.com and VirtualService with testing.example.com is the correct configuration?
> Gateway with testing.example.com and VirtualService with *.example.com is the correct configuration?
these will both work, the wildcard can be on either (or both)

Reply:
this is done, I am using the host package for comparing the `Host` in `Gateway` and `VirtualService`